submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/submitter/request/SubmissionRequestIndexer.kt

```diff
@@ -0,0 +1,44 @@
+package ac.uk.ebi.biostd.submission.submitter.request
+
+import ac.uk.ebi.biostd.persistence.common.model.RequestStatus.INDEXED
+import ac.uk.ebi.biostd.persistence.common.model.SubmissionRequestFile
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionRequestFilesPersistenceService
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionRequestPersistenceService
+import ebi.ac.uk.extended.model.ExtFile
+import ebi.ac.uk.extended.model.ExtSubmission
+import mu.KotlinLogging
+import uk.ac.ebi.extended.serialization.service.ExtSerializationService
+import uk.ac.ebi.extended.serialization.service.fileSequence
+import java.util.concurrent.atomic.AtomicInteger
+
+private val logger = KotlinLogging.logger {}
+
+class SubmissionRequestIndexer(
```

Comment:
now as it looks in the code we are indexing for processing only as releasing (publish in FTP) in mechanism does not use that. I think we need a todo for that part too.